src/NearestNeighbors.jl

```diff
@@ -22,14 +23,15 @@ export Euclidean,
 # Change this to enable debugging
 const DEBUG = false
 
-abstract NNTree{T <: AbstractFloat, P <: Metric}
+abstract NNTree{V <: AbstractVector, P <: Metric}
+
+typealias DistanceType Float64
+typealias MinkowskiMetric Union{Euclidean, Chebyshev, Cityblock, Minkowski}
 
-function check_input(tree::NNTree, points::AbstractArray)
-    ndim_points = size(points,1)
-    ndim_tree = size(tree.data, 1)
-    if ndim_points != ndim_tree
+function check_input{V1, V2}(tree::NNTree{V1}, points::Vector{V2})
+    if length(V1) != length(V2)
```

Comment:
Here and elsewhere you are allowing `V` to be `V <: AbstractVector`, and then calling `length()` on the type. Do you think `V <: StaticVector` is more appropriate?